src/pages/Lending/hooks/useRepaymentLockData.tsx

```diff
@@ -0,0 +1,92 @@
+import type { AccountId, AssetId } from '@shapeshiftoss/caip'
+import { useQuery } from '@tanstack/react-query'
+import axios from 'axios'
+import { getConfig } from 'config'
+import { useMemo } from 'react'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { getThorchainLendingPosition } from 'state/slices/opportunitiesSlice/resolvers/thorchainLending/utils'
+
+type UseLendingPositionDataProps = {
+  accountId?: AccountId
+  assetId?: AssetId
+}
+
+// non-exhaustive, we only use this to get the current blockheight
+type ThorchainBlock = {
+  header: {
+    height: number
+  }
+}
+
+export const useRepaymentLockData = ({ accountId, assetId }: UseLendingPositionDataProps) => {
+  const repaymentLockQueryKey: [
+    string,
+    { accountId: AccountId | undefined; assetId: AssetId | undefined },
+  ] = useMemo(() => ['thorchainLendingRepaymentLock', { accountId, assetId }], [accountId, assetId])
+
+  const repaymentLockData = useQuery({
+    staleTime: Infinity,
+    queryKey: repaymentLockQueryKey,
+    queryFn: async () => {
+      const daemonUrl = getConfig().REACT_APP_THORCHAIN_NODE_URL
+
+      const maybePosition = await (async () => {
+        if (!(accountId && assetId)) return null
+        const position = await getThorchainLendingPosition({ accountId, assetId })
+        return position
+      })()
+
+      const maybeBlockHeightPromise = (async () => {
+        if (!maybePosition) return null
+        const { data: block } = await axios.get<ThorchainBlock>(`${daemonUrl}/lcd/thorchain/block`)
+        const blockHeight = block.header.height
+        return blockHeight
+      })()
+
+      const mimirPromise = axios.get<Record<string, unknown>>(`${daemonUrl}/lcd/thorchain/mimir`)
+
+      const [maybeBlockHeight, { data: mimir }] = await Promise.all([
+        maybeBlockHeightPromise,
+        mimirPromise,
+      ])
+      // TODO(gomes): this is the repayment lock of the pool - not the borrower's
+      // we will want to make it programmatic in case there's an active position.
+      // https://dev.thorchain.org/thorchain-dev/lending/quick-start-guide
```

Comment:
not applicable anymore? we are getting it dynamically from mimir?

Reply:
b4055aa5aa79bd13394411d42cc37f5c985f34cf